feat: add commission fields matching RPC spec

R-K-H opened the pull request:
#### Problem
The `@solana/web3.js` SDK `InflationReward` type and `rewards` in the `meta` for `transaction status metadata object` properties do not match the JSON RPC spec in that they are missing the `commission` field. This is causing an issue with the types when attempting to access field value.

See:
- https://docs.solana.com/developing/clients/jsonrpc-api#getinflationreward
- https://docs.solana.com/developing/clients/jsonrpc-api#getblock

#### Summary of Changes
Adds the commission field in the `src/connection.ts`
- `InflationReward`
- `RewardsResult`
- `GetInflationRewardResult`
- `BlockResponse`
- `ParsedBlockResponse`
- `VersionedBlockResponse`
- `ConfirmedBlock`

Adds commission field in the `mockRpcResponse` in the `tests/connection.test.ts` file.

steveluscher reviewed (approved):
![giphy-1](https://user-images.githubusercontent.com/13243/210017301-084624df-e7ae-4fb8-a618-277bbca44a00.gif)

steveluscher reviewed (dismissed):
Looks like you have some Typescript errors. Want to fix those up and resubmit, @R-K-H?

R-K-H commented:
Sure thing, I've updated the PR and refactored the previous git commit message to pass lint there as well. I've ran the tests locally with one error from VoteProgram on live test (unrelated I think). 

Aside: Looks like it didn't like `optional()` and since `undefined` wasn't in the design pattern of current implementation, I just went ahead and continued the pattern for other undefined fields coming from the RPC spec with the null being acceptable.

steveluscher commented on web3.js/src/connection.ts:
```diff
@@ -767,6 +767,8 @@ export type InflationReward = {
   /** post balance of the account in lamports */
   postBalance: number;
+  /** vote account commission when the reward was credited */
+  commission: number | null;
```
Are you sure this is right? The [JSON-RPC docs](https://docs.solana.com/developing/clients/jsonrpc-api#results-20) describe this property as `commission: <u8|undefined>`.

```suggestion
  commission?: number;
```

R-K-H replied:
You're correct! Will fix.

steveluscher replied:
I mean, I'm _really_ just asking. It's a distinct possibility that the docs are wrong. https://github.com/solana-labs/solana/pull/27655

R-K-H replied:
So it is returning null from the request, when doing the live testing it does seem to require that the field is nullable. And I know from my own use that making the request to mainnet the JSON response is returning the commission (at least for InflationReward) as a number.

But I'm fixing right now which I suppose supporting the `optional(nullable(number()))` more closely matches the JSON RPC spec with it being ABLE to be undefined (although I'm not certain it does return without the field).

steveluscher replied:
Allowing for `undefined` would definitely protect against a response from an old RPC fataling the client. I don't think there are any RPCs old enough to return `undefined` for this property but ¯\_(°ペ)_/¯

R-K-H replied:
Indeed and agreed. Updated.
